filter/auth_ext.go

```diff
@@ -0,0 +1,24 @@
+package filter
+
+import (
+	"github.com/apache/dubbo-go/common"
+	"github.com/apache/dubbo-go/protocol"
+)
+
+type AccessKeyPair struct {
+	AccessKey    string `yaml:"accessKey"   json:"accessKey,omitempty" property:"accessKey"`
+	SecretKey    string `yaml:"secretKey"   json:"secretKey,omitempty" property:"secretKey"`
+	ConsumerSide string `yaml:"consumerSide"   json:"ConsumerSide,consumerSide" property:"consumerSide"`
+	ProviderSide string `yaml:"providerSide"   json:"providerSide,omitempty" property:"providerSide"`
+	Creator      string `yaml:"creator"   json:"creator,omitempty" property:"creator"`
+	Options      string `yaml:"options"   json:"options,omitempty" property:"options"`
+}
+
+type Authenticator interface {
+	Sign(protocol.Invocation, *common.URL) error
+	Authenticate(protocol.Invocation, *common.URL) error
+}
+
+type AccesskeyStorage interface {
+	GetAccesskeyPair(protocol.Invocation, *common.URL) *AccessKeyPair
+}
```

Comment:
Move to filter_impl/auth/ folder?

Reply:
> Move to filter_impl/auth/ folder?

That is a good idea.

Reply:
If move to auth folder, will  issue circular dependencies problem.

Reply:
I found that the relevant interfaces of TPS `TpsLimitStrategy` and` TpsLimitStrategyCreator` are placed in the `filter` directory.

Reply:
Why issue circular dependencies? I only found it called by filter/filter_impl/auth/authenticator.go.

Reply:
Is there any other code not included in this pr?